sklearn/linear_model/logistic.py

```diff
@@ -652,7 +679,7 @@ def logistic_regression_path(X, y, pos_class=None, Cs=10, fit_intercept=True,
             le = LabelEncoder()
             Y_multi = le.fit_transform(y).astype(X.dtype, copy=False)
 
-        w0 = np.zeros((classes.size, n_features + int(fit_intercept)),
+        w0 = np.zeros((max(2, classes.size), n_features + int(fit_intercept)),
```

Comment:
Could you explain why we need this for changing the default solver? Or is this an unrelated fix?

Reply:
This is an unrelated fix, when only one class is given with `multi_class='multinomial'`.
This came up as I ran the tests with the future defaults.

Reply:
That makes no sense again. In binary logistic regression there should only be a single weight vector. I'm not sure what ovr vs multinomial means but maybe multinomial always uses n_classes many weight-vectors and that's wrong imho.